src/lib/transport/ip/udp_recv.c

```diff
@@ -359,10 +359,14 @@ static int ci_udp_recvmsg_get(ci_udp_recv_info* rinf, ci_iovec_ptr* piov)
   rc = oo_copy_pkt_to_iovec_no_adv(ni, pkt, piov, pkt->pf.udp.pay_len);
 
   if(CI_LIKELY( rc >= 0 )) {
+    if(CI_UNLIKELY( rc < pkt->pf.udp.pay_len )) {
 #if HAVE_MSG_FLAGS
-    if(CI_UNLIKELY( rc < pkt->pf.udp.pay_len && msg != NULL ))
-      rinf->msg_flags |= LOCAL_MSG_TRUNC;
+      if( msg != NULL )
+        rinf->msg_flags |= LOCAL_MSG_TRUNC;
 #endif
+      if( rinf->flags & MSG_TRUNC )
+        rc = pkt->pf.udp.pay_len;
+    }
```

Comment:
It looks like all these code should be under `#if HAVE_MSG_FLAGS`.

Reply:
Ok I changed this. I'm not familiar with the codebase enough to know how this code is reachable in kernel mode to make a judgement myself on whether it should be wrapped in HAVE_MSG_FLAGS.